### PR TITLE
Support sending beacons to multiple backends

### PR DESCRIPTION
**Pull request by gwangibm**

### CODE REVIEW

File: lib/transmission/batched.js [hunk 1]
```diff
@@ -59,12 +60,35 @@ function transmit() {
...
+  }
+
+  if (!vars.reportingBackends || vars.reportingBackends.length === 0) {
+    pendingBeacons.length = 0;
+    return;
+  }
+
+  for (let i = 0, len = vars.reportingBackends.length; i < len; i++) {
+    let reportingBackend: ReportingBackend = vars.reportingBackends[i];
+    if (hasOwnProperty(reportingBackend, 'reportingUrl')) {
```
Comment (zhouhn-ibm) [inline lib/transmission/batched.js:76]: Question: Do we need to validate the URL during each transmition? Maybe we should validate and reject incorrect "reportingBackend" during command api.

Comment (gwangibm) [reply lib/transmission/batched.js:76]: `reportingUrl` does not do any validation as well. I guess do a simple malformed url check might not add much value, so just leave it to the final transmission request to report any error.

File: lib/ignoreRules.js [hunk 2]
```diff
@@ -1,5 +1,6 @@
 // @flow
 
+import { isTransmitionRequest } from './transmission/util';
```
Comment (lisr) [inline lib/ignoreRules.js:3]: The old code format do not have extra spaces between `{}`

File: lib/transmission/batched.js (hunk 1 above)
Comment (zhouhn-ibm) [reply lib/transmission/batched.js:76]: Let me rephrase. Here I mean do we need to check this `hasOwnProperty` during each transmission? If the input reportingBackend does not have 'reportingUrl', the command api can reject it at first place and keep vars.reportingBackend empty.

**Comment by zhouhn-ibm** [comment]
Question: With "reportingBackend", does end user still need to call ineum(key)?

**Comment by gwangibm** [comment]
> Question: With "reportingBackend", does end user still need to call ineum(key)?

`ineum(key)` will be ignored.

File: lib/transmission/batched.js [hunk 3]
```diff
@@ -59,12 +60,37 @@ function transmit() {
...
+      if (i > 0) {
```
Comment (zhouhn-ibm) [inline lib/transmission/batched.js:77]: Here we replace key from the second reportingBackend. So we assume the first reportingBackend has the same key as ineum(key), right?

Comment (gwangibm) [reply lib/transmission/batched.js:77]: No, the key of pendingBeacons were set to the key of the first reportingBackend, see ln19 at https://github.com/instana/weasel/blob/multi-backends/lib/commonBeaconProperties.js

Comment (zhouhn-ibm) [reply lib/transmission/batched.js:77]: OK, I see.

**Review by zhouhn-ibm — APPROVED** [review]
LGTM
